os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -473,6 +477,17 @@ trwifi_result_e wifi_netmgr_utils_deinit(struct netdev *dev)
 	if (ret == RTK_STATUS_SUCCESS) {
 		g_mode = RTK_WIFI_NONE;
 		wuret = TRWIFI_SUCCESS;
+		rtw_mutex_get(&scanlistbusy);
+		if (scan_timer.timer_hdl != NULL) {
+			rtw_cancel_timer(&(scan_timer));
+			rtw_del_timer(&(scan_timer));
+		}
+		if (saved_scan_list) {
+			rtw_mfree((unsigned char *)saved_scan_list, sizeof(ap_scan_list_s) * scan_number);
+			saved_scan_list = NULL;
+		}
+		scan_number = 0;
+		rtw_mutex_put(&scanlistbusy);
 		rtw_mutex_free(&scanlistbusy);
 	} else {
 		ndbg("[RTK] Failed to stop STA mode\n");

```

Comment:
If deinit were failed then is it ok not to release wi-fi resources like timer scan list?
Could you tell me what behavior can be happened when driver was failed to deinit?

Reply:
@pillip8282, 
Please find the scan related resources behavior,
1) timer: scan_timer
The timer init during scan(save_scan_list()) called for the first time. 
It will be deleted during timeout/scan list malloc fail and wifi deinit.
So even if the wifi deinit is unsuccessful, the timer will be deleted during timeout.

2) scan list: saved_scan_list
The scan list will be freed once the timer expires(_scan_timer_handler()) or application calls scan again before timeout(save_scan_list()).
It will also be freed after successful wifi deinit.
So even if the wifi deinit is unsuccessful, the scan list will be freed during timeout.

3) mutex: scanlistbusy
The mutex init after success at "wifi_netmgr_utils_init()" and free after success at "wifi_netmgr_utils_deinit()".
Even if the wifi deinit is failed, all other operations which consists of mutex "get" is followed by mutex "put". 
So mutex can wait to be freed until wifi deinit is successful. 
For next succesful wifi init(which also init the mutex), previous wifi deinit has to be successful.